indexer-api/src/infra/api/v3.rs

```diff
@@ -109,6 +111,65 @@ where
 
 impl<T> AsBytesExt for T where T: AsRef<[u8]> {}
 
+/// Wrapper around a Bech32-encoded Cardano reward address.
+#[derive(Debug, Display, Clone, PartialEq, Eq, Hash, Serialize, Deserialize, AsRef)]
+#[debug("{_0}")]
+#[as_ref(str)]
+pub struct CardanoRewardAddress(String);
+
+scalar!(CardanoRewardAddress);
+
+impl CardanoRewardAddress {
+    /// Decode this Bech32 Cardano reward address into a CardanoRewardAddress.
+    pub fn decode(&self) -> Result<DomainCardanoRewardAddress, Bech32DecodeError> {
+        decode_cardano_reward_address(&self.0).map_err(Bech32DecodeError::Decode)
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum Bech32DecodeError {
+    #[error("cannot bech32-decode Cardano reward address")]
+    Decode(#[from] DecodeCardanoRewardAddressError),
+}
+
+impl TryFrom<String> for CardanoRewardAddress {
+    type Error = DecodeCardanoRewardAddressError;
+
+    fn try_from(s: String) -> Result<Self, Self::Error> {
+        // Validate by attempting to decode.
+        decode_cardano_reward_address(&s)?;
+        Ok(Self(s))
+    }
+}
+
+impl TryFrom<&str> for CardanoRewardAddress {
+    type Error = DecodeCardanoRewardAddressError;
+
+    fn try_from(s: &str) -> Result<Self, Self::Error> {
+        // Validate by attempting to decode.
+        decode_cardano_reward_address(s)?;
+        Ok(Self(s.to_owned()))
+    }
+}
+
+/// Extension trait for encoding CardanoRewardAddress to Bech32 format.
+pub trait CardanoRewardAddressExt {
```

Comment:
Why is this implemented as an "extension"? Wouldn't it be better to implement is "simply" on the `CardanoRewardAddress`?